src/libraries/System.Text.Json/System.Text.Json.sln

```diff
@@ -35,6 +35,20 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "ref", "ref", "{EC8CE194-261
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{676B6044-FA47-4B7D-AEC2-FA94DB23A423}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "source_generation", "source_generation", "{A403699B-1BB2-4B23-AE05-97608C23EB27}"
```

Comment:
nit: the "source_generation" name is probably fine since the terminology is "source generators". However, was there precedence for this naming and\or were other names considered or perhaps attempts at avoiding underscores? I don't know if there is a best practice to avoid underscores for project names.
- "sourceGeneration"
- "srcGen"

Reply:
I tried to go for the most descriptive thing. I didn't see any precedent for camel case naming, but I did see "source_package"; hence "source_generation".

Per https://github.com/dotnet/runtime/pull/51300#discussion_r614213341, @maryamariyan has the logging generator under "gen" & is establishing a pattern for including the project when building our libraries. We should use the thing same here pending opinions from @safern, @ViktorHofer. cc @ericstj

Reply:
I don't have much opinion other than a request for consistency. @eerhardt or @stephentoub might have an opinion.

Reply:
I used `gen`, and I'm including the project to src.proj in the same way as https://github.com/dotnet/runtime/pull/51064.